### PR TITLE
[WIP]  Enhancements in feeds organization recognition API

### PR DESCRIPTION
**Pull request by yatishTrootech**
Description [Here](https://github.com/rewardz/skor/pull/7245#issue-2178830752)

### CODE REVIEW

File: feeds/serializers.py [hunk 1]
```diff
@@ -63,6 +66,54 @@ def get_info_for_greeting_post(post):
...
+    """
+    Returns the serialized images
+    :params: post: Post
+    :returns: [ImagesSerializer]
+    """
+    all_images = list()
+    if not post.post_type == POST_TYPE.USER_CREATED_POLL:
+        all_images = ImagesSerializer(post.images_set, many=True, read_only=True).data
+    if post.ecard:
+        all_images.insert(0, ECardSerializer(post.ecard).data)
```
Comment (dryice) [inline feeds/serializers.py:79]: not very sure here: `ImageSerializer` and `ECard;serializer` are in 2 different formats. Will FE/Mobile be confused when we combine them into the same list?

File: feeds/serializers.py [hunk 2]
```diff
@@ -63,6 +66,54 @@ def get_info_for_greeting_post(post):
...
+    return all_images
+
+
+def get_user_strength(instance):
+    """
+    Returns the serialized user strength data
+    :params: instance: Post
+    :returns: UserStrengthSerializer/None
+    """
+    transaction = instance.transactions.first()
```
Comment (dryice) [inline feeds/serializers.py:89]: probably need more filters here? E.g, if the `first` call gets a redemption transaction, we won't be able to return the user strength?

File: feeds/serializers.py (hunk 1 above)
Comment (yatishTrootech) [reply feeds/serializers.py:79]: Not sure on this, Just moved same logic to function, @suyeshkandpal28 can you please confirm ?

File: feeds/serializers.py (hunk 2 above)
Comment (yatishTrootech) [reply feeds/serializers.py:89]: Same here as well moved this same code from serializer to function, Just not to break existing logic and response

Comment (dryice) [reply feeds/serializers.py:89]: can we verify it? E.g, create a user with only one appreciation transaction and lots of redemption transactions, and see what happens?

**Review by dryice — APPROVED** [review]
LGTM. Thanks!
The 2 comments can be dealt later cause they match existing code thus won't cause new issues.

**Review by rishireward — APPROVED** [review]

**Pull request merged**